src/main/kotlin/com/demonwav/mcdev/i18n/I18nAnnotator.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2017 minecraft-dev
+ *
+ * MIT License
+ */
+
+package com.demonwav.mcdev.i18n
+
+import com.demonwav.mcdev.i18n.intentions.RemoveDuplicatesIntention
+import com.demonwav.mcdev.i18n.intentions.RemoveUnmatchedPropertyIntention
+import com.demonwav.mcdev.i18n.intentions.TrimKeyIntention
+import com.demonwav.mcdev.i18n.lang.gen.psi.I18nProperty
+import com.intellij.lang.annotation.AnnotationHolder
+import com.intellij.lang.annotation.Annotator
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiElement
+
+class I18nAnnotator : Annotator {
+    override fun annotate(psiElement: PsiElement, annotationHolder: AnnotationHolder) {
+        if (psiElement is I18nProperty) {
+            checkPropertyMatchesDefault(psiElement, annotationHolder)
+            checkPropertyKey(psiElement, annotationHolder)
+            checkPropertyDuplicates(psiElement, psiElement.parent.children, annotationHolder)
+        }
+    }
+
+    private fun checkPropertyKey(property: I18nProperty, annotations: AnnotationHolder) {
+        if (property.key != property.trimmedKey) {
+            val range = TextRange.from(property.textRange.startOffset, property.key.length)
+            annotations.createWarningAnnotation(range, "Translation key contains whitespace at start or end")
+                .registerFix(TrimKeyIntention())
+        }
+    }
+
+    private fun checkPropertyDuplicates(property: I18nProperty, siblings: Array<PsiElement>, annotations: AnnotationHolder) {
+        val count = siblings.count { it is I18nProperty && property.key == it.key }
+        if (count > 1) {
+            annotations.createWarningAnnotation(property, "Duplicate translation keys \"${property.key}\"")
+                .registerFix(RemoveDuplicatesIntention(property))
+        }
+    }
+
+    private fun checkPropertyMatchesDefault(property: I18nProperty, annotations: AnnotationHolder) {
+        for (prop in property.project.findDefaultProperties(domain = I18nElementFactory.getResourceDomain(property.containingFile.virtualFile))) {
```

Comment:
That looks like 

```kotlin
if (property.project.findDefaultProperties(domain = I18nElementFactory.getResourceDomain(property.containingFile.virtualFile).any { prop.key == property.key }) { 
    return
}
```

Not sure if that is nicer, just thought I'd mention it. Feel free to keep it as-is